pkg/cmd/generate/main.go

```diff
@@ -0,0 +1,111 @@
+package generate
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	k8s_json "k8s.io/apimachinery/pkg/runtime/serializer/json"
+	"k8s.io/apimachinery/pkg/util/yaml"
+
+	v1 "github.com/jaegertracing/jaeger-operator/pkg/apis/jaegertracing/v1"
+	"github.com/jaegertracing/jaeger-operator/pkg/cmd/start"
+	"github.com/jaegertracing/jaeger-operator/pkg/strategy"
+)
+
+// NewGenerateCommand starts the Jaeger Operator
+func NewGenerateCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "generate",
+		Short: "Generate YAML manifests from Jaeger CRD",
+		Long: `Generate YAML manifests from Jaeger CRD.
+
+Defaults to reading Jaeger CRD from standard input and writing the manifest file to standard output, override with --cr <filename> and --output <filename>.`,
+		RunE: generate,
+	}
+
+	cmd.Flags().String("cr", "/dev/stdin", "Input Jaeger CRD")
+	viper.BindPFlag("cr", cmd.Flags().Lookup("cr"))
+
+	cmd.Flags().String("output", "/dev/stdout", "Where to print the generated YAML documents")
+	viper.BindPFlag("output", cmd.Flags().Lookup("output"))
+
+	start.AddGeneratorFlags(cmd)
+
+	return cmd
+}
+
+func createSpecFromYAML(filename string) (*v1.Jaeger, error) {
+	// #nosec   G304: Potential file inclusion via variable
+	f, err := os.Open(filename)
+	if err != nil {
+		return nil, err
+	}
+	defer f.Close()
+
+	var spec v1.Jaeger
+	decoder := yaml.NewYAMLOrJSONDecoder(f, 8192)
+	if err := decoder.Decode(&spec); err != nil && err != io.EOF {
+		return nil, err
+	}
+
+	return &spec, nil
+}
+
+func generate(cmd *cobra.Command, args []string) error {
```

Comment:
nit: `func generate(_ *cobra.Command, _ []string)` it's easier to read that the arguments are not used.